angular/src/lib/modal/modal-ref.ts

```diff
@@ -0,0 +1,39 @@
+import { OverlayRef } from '@angular/cdk/overlay';
+import { Subject } from 'rxjs';
+import { TemplateRef, Type } from '@angular/core';
+
+interface ModalCloseEvent {
+   type: 'backdropClick' | 'close';
+   data: object;
+}
+
+export type ModalContent = TemplateRef<any> | Type<any>;
+
+export class ModalRef {
+  private onHide = new Subject<ModalCloseEvent>();
+  onHide$ = this.onHide.asObservable();
+
+  constructor(public overlay: OverlayRef,
+              public content: ModalContent,
+              public data: object,
+              public backdropClickExit: boolean) {
```

Comment:
Can we tab in the constructor parameters just so we can keep the style consistent across components 
```
constructor(
    public content: ModalContent,
    public data: object,
    .....
  ) {} ```